integr8ly/integreatly-operator
# MGDAPI-4814 - Race condition setting CRO network strategy

# Issue link
https://issues.redhat.com/browse/MGDAPI-4814

# What
There is a race condition in RHOAM addon install where the CRO network strategy might not be updated until after the installation has already began. We saw this in the [pipeline run here](https://master-jenkins-csb-intly.apps.ocp-c1.prod.psi.redhat.com/job/ManagedAPI/job/managed-api-install-addon-flow/862/console).

Please see more details in Jira: https://issues.redhat.com/browse/MGDAPI-4814

# HOW
Move the CIDR block update to after the create / update of the strategy map
Ensure that if the addon parameter CIDR range is set, that we do not create the CRO subscription until the strategy is present with the CIDR range set

# Verification steps
- Create ccs cluster
- Create private Pipeline in playground, as copy from managed-api-install-addon-flow pipeline.
- Check that Pipeline passed successfully
